GrandCentralBoard/Widgets/Harvest/View/AreaBarChartView.swift

```diff
@@ -0,0 +1,136 @@
+//
+//  Created by Oktawian Chojnacki on 13.04.2016.
+//  Copyright © 2016 Oktawian Chojnacki. All rights reserved.
+//
+
+import UIKit
+import GrandCentralBoardCore
+
+
+private let FirstScaleLineHeight: CGFloat = 12
+private let NormalScaleLineHeight: CGFloat = 4
+private let LastScaleLineHeight: CGFloat = 32
+
+final class AreaBarHorizontalAxisStackView : UIStackView {
+
+    override func awakeFromNib() {
+        super.awakeFromNib()
+
+        axis = .Horizontal
+        distribution = .EqualSpacing
+        alignment = .Top
+    }
+
+    func drawAxisWithViewModel(viewModel: AreaBarChartViewModel) {
+
+        let stops = viewModel.horizontalAxisStops
+
+        for stop in 0...stops {
+
+            let scaleLineView = UIView()
+            scaleLineView.backgroundColor = UIColor.axisGray()
+
+            addArrangedSubview(scaleLineView)
+
+            let height = scaleLineHeightForStop(stop, stops: stops)
+            scaleLineView.heightAnchor.constraintEqualToConstant(height).active = true
+
+            scaleLineView.widthAnchor.constraintEqualToConstant(1).active = true
+        }
+    }
+
+    private func scaleLineHeightForStop(stop: Int, stops: Int) -> CGFloat {
+        switch stop {
+        case 0:
+            return FirstScaleLineHeight
+        case stops:
+            return LastScaleLineHeight
+        default:
+            return NormalScaleLineHeight
+        }
+    }
+}
+
+final class AreaBarChartView : UIView, ViewModelRendering {
+    typealias ViewModel = AreaBarChartViewModel
+
+    @IBOutlet private var barStackView: AreaBarStackView!
+    @IBOutlet private var axisStackView: AreaBarHorizontalAxisStackView!
+
+    @IBOutlet private var horizontalAxisLabel: UILabel!
+    @IBOutlet private var hotizontalAxisCountLabel: UILabel!
+
+    override func awakeFromNib() {
+        super.awakeFromNib()
+
+        translatesAutoresizingMaskIntoConstraints = false
+    }
+
+    // MARK: ViewModelRendering
+
+    private(set) var state: RenderingState<ViewModel> = .Waiting {
+        didSet { handleTransitionFromState(oldValue, toState: state) }
+    }
+
+    func render(viewModel: ViewModel) {
+        state = .Rendering(viewModel)
+    }
+
+    func failure() {
+        state = .Failed
+    }
+
+    // MARK: - Transitions
+
+    private func handleTransitionFromState(state: RenderingState<ViewModel>, toState: RenderingState<ViewModel>) {
+        switch (state, toState) {
+        case (_, .Rendering(let viewModel)):
```

Comment:
If you're ignoring the first field anyway, maybe you could switch just on `toState`